File: 02-programacion-basica/10-colecciones-basicas/java/ejemplos-guiados/ColeccionesBasicas.java
import java.util.ArrayList;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Map;
import java.util.Set;

public class ColeccionesBasicas {
    public static void main(String[] args) {
        // List conserva una secuencia y puede contener elementos repetidos.
        List<String> cursos = new ArrayList<>(List.of("Java", "Scala", "Java"));

        // Set representa unicidad; los duplicados de la lista desaparecen.
        Set<String> cursosUnicos = new HashSet<>(cursos);

        // Map asocia una clave con un valor.
        Map<String, Integer> avance = new HashMap<>();
        avance.put("Java", 70);
        avance.put("Scala", 25);

        System.out.println("Lista: " + cursos);
        System.out.println("Sin duplicados: " + cursosUnicos);
        System.out.println("Avance Java: " + avance.get("Java") + "%");
    }
}

File: 02-programacion-basica/10-colecciones-basicas/scala/ejemplos-guiados/ColeccionesBasicas.scala
object ColeccionesBasicas:
  def main(args: Array[String]): Unit =
    // Las colecciones literales de este ejemplo son inmutables.
    val cursos = List("Java", "Scala", "Java")
    val cursosUnicos = cursos.toSet
    val avance = Map("Java" -> 70, "Scala" -> 25)

    println(s"Lista: $cursos")
    println(s"Sin duplicados: $cursosUnicos")
    println(s"Avance Java: ${avance("Java")}%")
